src/lib/socket.c

```diff
@@ -2392,16 +2400,64 @@ static BIO_METHOD bio_methods = {
 	NULL,
 };
 
+static BIO_METHOD* get_bio_methods() {
```

Comment:
This should probably be so_get_bio_methods, or so_getbio_methods. socket.c might be #include'd into other code.  

One of the current bugs preventing creation of a static archive is linkage. The simplest solution is to just build a single source file, with every routine declared static except for luaopen routines. That requires minding namespaces. The alternative is to change the default visibility to non-exported using special GCC and clang flags.

I'll probably go with the alternative, but it's definitely messier from a build perspective. And going with a single source file build also would simplify the Makefile.